helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -367,22 +367,28 @@ public void increaseMessageReceived(List<Message> messages) {
   }
 
   /**
-   * Update max capacity usage for per instance. Before calling this API, we assume the instance
-   * monitors are already registered in ReadClusterDataStage. If the monitor is not registered, this
-   * max usage update will fail.
+   * Update instance capacity status for per instance, including max usage and capacity of each
+   * capacity key. Before calling this API, we assume the instance monitors are already registered
+   * in ReadClusterDataStage. If the monitor is not registered, this instance capacity status update
+   * will fail. Keys for maxUsageMap and capacityMap should be the same.
    *
-   * @param maxUsageMap a map of max capacity usage, {instance: maxCapacityUsage}
+   * @param maxUsageMap A map of max capacity usage, {instance: maxCapacityUsage}
+   * @param capacityMap A map of instance capacity, {instance: {capacity key: capacity value}}
    */
-  public void updateInstanceMaxUsage(Map<String, Double> maxUsageMap) {
+  public void updateInstanceCapacityStatus(Map<String, Double> maxUsageMap,
```

Comment:
I combine max usage and instance capacity update into one api, in the purpose of to avoid one more for loop and synchronized lock for instance capacity update. The downside is, we assume keys (instance names) are the same. If you think we should separate the apis, please comment.